tests/test_self.py

```diff
@@ -741,6 +741,87 @@ def test_fail_under(self):
             code=22,
         )
 
+    def test_fail_on(self):
+        """Run same test cases as --fail-under, but run with/without a detected issue code"""
+        test_cases = [
```

Comment:
We should use ``@pytest.mark.parametrize`` for those.

Reply:
Updated, cheers!